app/assets/src/components/AlignmentViz.jsx

```diff
@@ -0,0 +1,276 @@
+    }
+
+    let aligned_portion = sequence.slice(readInfo.metrics[4] - 1, readInfo.metrics[5])
+    let left_portion = (readInfo.metrics[4] - 2) >= 0 ? sequence.slice(0, readInfo.metrics[4] - 1) : ""
+    let right_portion = readInfo.metrics[5] < seq_len ? sequence.slice(readInfo.metrics[5]) : ""
+
+    if (readInfo.refInfo[0].length > left_portion.length) {
+      // trim refInfo[0]
+      readInfo.refInfo[0] = readInfo.refInfo[0].slice(readInfo.refInfo[0].length - left_portion.length)
+    } else {
+      // pad refInfo[0]
+      while(readInfo.refInfo[0].length < left_portion.length) {
+        readInfo.refInfo[0] = ' ' + readInfo.refInfo[0]
+      }
+    }
+
+    if (readInfo.refInfo[2].length > right_portion.length) {
+      //trim refInfo[2]
+      readInfo.refInfo[2] = readInfo.refInfo[2].slice(0, right_portion.length)
+    } else {
+      while(readInfo.refInfo[2].length < right_portion.length) {
+        readInfo.refInfo[2] += ' '
+      }
+    }
+    let [quality_string1, mis_matches1] = this.generateQualityString(readInfo.refInfo[1], aligned_portion)
+    let [quality_string2, mis_matches2] = this.generateQualityString(readInfo.refInfo[1], this.complementSeq(aligned_portion))
+    let quality_string = mis_matches1 < mis_matches2 ? quality_string1 : quality_string2
+    let white_space_left = this.repeatStr(' ', left_portion.length)
+    let white_space_right = this.repeatStr(' ', right_portion.length)
+    let ref_seq_display = readInfo.refInfo.join("|")
+    let read_seq_display = [left_portion, aligned_portion, right_portion].join("|")
+    let quality_string_display = [white_space_left, quality_string, white_space_right].join("|")
+
+    // generate quality string
+    return {
+      reversed: reversed,
+      read_part: readPart,
+      ref_display: ref_seq_display,
+      read_display: read_seq_display,
+      quality_display: quality_string_display
+    }
+  }
+
+  repeatStr(c, n) {
+    let output = ''
+    for (let i = 0; i < n; i += 1) {
+      output += c
+    }
+    return output
+  }
+
+  generateQualityString(refString, seqString) {
+    let mis_matches = 0
+    let quality_string = ''
+    for (let i = 0; i < seqString.length; i += 1) {
+      let cr = refString[i]
+      let cs = seqString[i]
+      if (cr === 'N' || cs === 'N' || cr === cs) {
+        quality_string += ' '
+      } else {
+        quality_string += 'X'
+        mis_matches += 1
+      }
+    }
+    return [quality_string, mis_matches]
+  }
+
+  complementSeq(seq) {
+    let output = ''
+    for (let c of seq) {
+      switch (c) {
+        case 'A':
+          output += 'T';
+          break;
+        case 'T':
+          output += 'A';
+          break;
+        case 'C':
+          output += 'G';
+          break;
+        case 'G':
+          output += 'C';
+          break;
+        default:
+          output += c
+      }
+    }
+    return output
+  }
+
+  render() {
+    const nopadding = {padding: '0px'}
```

Comment:
ideally would start another .scss file?

Reply:
great. I was doing inline until we have a design.